index.js

```diff
@@ -148,6 +154,35 @@ function readStream (stream, encoding, length, limit, callback) {
   var complete = false
   var sync = true
 
+  var done = preserveAsyncContext('RawBodyDone', function () {
```

Comment:
I have a couple questions here:

1. Does the entire `done` function need to be what is wrapped, or would just wrapping the provided `callback` be all that is necessary?
2. Does the name (`RawBodyDone`) have any external visibility / meaning?

Reply:
1. Yes, that's a good idea, wrapping only a `callback` should be sufficient and result in an even smaller change to this module :slightly_smiling_face: 
2. Yes, it will be provided to anyone that will register an async hook with `init` callback as a `type` argument whenever this function is invoked. More info: https://nodejs.org/api/async_hooks.html#async_hooks_type

Reply:
Sorry about this pr; you reminded me about it in the other one :) for the resource name, do you think it should be prefixed with this module's name? This module's name has a dash in it, withthe dashless name being a different module: https://www.npmjs.com/package/rawbody

Reply:
That's true, but if we apply dash to upper camel case conversion, `raw-body` becomes `RawBody` while `rawbody` becomes `Rawbody`.